Data Structures/matrix_sum_query.cpp

```diff
@@ -0,0 +1,68 @@
+// Name : Shivam Singhal
+// Email ID : shivamsinghal0610@gmail.com
+// Algorithm to calculate the sum of elements contained in a sub-matrix
+// formed by given 2 co-ordinates in a query taking first point as upper left
+// corner and the second point as the lower right corner
+
+#include <iostream>
+using namespace std;
+
+int main()
+{
+	int m, n, q;
+	cin >> m >> n;
+	cout << "Dimensions of matrix are : " << m <<", "<<n<<endl;
+	int a[m][n];
+	for(int i=0; i<m; i++){
+		for(int j=0; j<n; j++)
+			cin >> a[i][j];
+	}
+	cout << "Matrix elements are :" << endl;
+	for(int i=0; i<m; i++){
+		for(int j=0; j<n; j++)
+			cout << a[i][j] << " ";
+		cout << endl;
+	}
+	for(int i=1; i<m; i++){
+		for(int j=0; j<n; j++){
+			a[i][j] += a[i-1][j];
```

Comment:
Write comments to explain what sum are you taking everywhere in the program.

Reply:
Okay, sure.